pkg/services/notary/notary.go

```diff
@@ -376,15 +390,24 @@ func (n *Notary) PostPersist() {
 	}
 
 	n.reqMtx.Lock()
+	// @roman-khimov, an option is to copy n.requests in order not to hold lock over n.reqMtx for the whole cycle,
+	// but we need to take a write lock over every request anyway since it's modified by n.finalize.
+	// We also need to clean the list of requests with 0 fallbacks, this requires n.reqMtx to be taken.
```

Comment:
Let's try it this way, the cycle itself is rather small, so shouldn't be a big problem.